lib/order.rb

```diff
@@ -9,10 +13,73 @@ def initialize(id, products)
 
     def total
       # TODO: implement total
+      total = 0
+      products.each {|name, price| total += price}
+      total = total + (total* 0.075).round(2)
     end
 
     def add_product(product_name, product_price)
       # TODO: implement add_product
+      if !@products.has_key? product_name
+        @products[product_name] = product_price
+        true
+      else
+        false
+      end
+    end
+
+    def remove_product(product_name)
+      @products.delete product_name
+    end
+
+    def product_keys
+      @products.keys
+    end
+
+    def self.cost_item_hash(info_array)
+      hashy_hash = {}
+      while info_array != []
+        item = info_array.delete_at(0)
+        cost = info_array.delete_at(0)
+        hashy_hash[item] = cost
+      end
+      hashy_hash
+    end #end
+
+    #reads csv file for order
+    def self.get_csv_info
+      CSV.open(CSVFILE, 'r')
+    end
+
+    def self.all
+      all_orders = []
+      (self.get_csv_info).each do |ugly_string_line|
+        #sort the ugly string out
+        #{id_number => {item => price}}
+
+        info_array = (ugly_string_line[1]).split(/;|:/)
+
+        id_num = (ugly_string_line[0]).to_i
```

Comment:
While I appreciate your use of parentheses here, I think I'd argue they don't make your code more readable than it would be without them. Something like

```ruby
info_array = ugly_string_line[1].split(/;|:/)
id_num = ugly_string_line[0].to_i
```

is just as clear (at least to my eye) as what you have. Moreover, the parentheses imply that there's something tricky or subtle going on there, when there really isn't.